app/forms/declaration/ConsignmentReferences.scala

```diff
@@ -16,24 +16,53 @@
 
 package forms.declaration
 
-import forms.{DeclarationPage, Ducr, Lrn}
+import forms.{DeclarationPage, Ducr, Lrn, Mrn}
 import models.viewmodels.TariffContentKey
-import models.DeclarationType.{CLEARANCE, DeclarationType}
+import models.DeclarationType.{CLEARANCE, DeclarationType, SUPPLEMENTARY}
 import play.api.data.{Form, Forms}
+import play.api.data.Forms.{optional, text}
 import play.api.libs.json.Json
+import utils.validators.forms.FieldValidator.{isMissing, isPresent}
 
-case class ConsignmentReferences(ducr: Ducr, lrn: Lrn)
+case class ConsignmentReferences(ducr: Ducr, lrn: Lrn, mrn: Option[Mrn] = None)
 
 object ConsignmentReferences extends DeclarationPage {
 
-  val mapping =
-    Forms.mapping("ducr" -> Ducr.ducrMapping, "lrn" -> Lrn.mapping("declaration.consignmentReferences.lrn"))(ConsignmentReferences.apply)(
-      ConsignmentReferences.unapply
-    )
-
   implicit val format = Json.format[ConsignmentReferences]
 
-  def form(): Form[ConsignmentReferences] = Form(mapping)
+  def form(decType: DeclarationType): Form[ConsignmentReferences] = {
+
+    def form2Model: (Ducr, Lrn, Option[Mrn]) => ConsignmentReferences = {
+      case (ducr, lrn, mrn) =>
+        mrn match {
+          case Some(x) => ConsignmentReferences(ducr, lrn, mrn)
+          case None    => ConsignmentReferences(ducr, lrn, None)
+        }
+    }
+
+    def model2Form: ConsignmentReferences => Option[(Ducr, Lrn, Option[Mrn])] =
```

Comment:
Maybe worth to have a proper `case class(ducr, lrn, mrn)` ?

Reply:
Not sure what you mean, this is the unapply function for the mapping (so it must return an Option[Tuple3])